nncf/config/extractors.py

```diff
@@ -11,64 +11,49 @@
-    :param config: An instance of the NNCFConfig.
-    :return: List of the NNCFConfigs, each of which contains specific algorithm
-        parameters related only to this algorithm
-    """
-    compression_config_json_section = config.get('compression', {})
-    compression_config_json_section = deepcopy(compression_config_json_section)
 
-    hw_config_type = None
-    target_device = config.get('target_device', 'ANY')
-    if target_device != 'TRIAL':
-        hw_config_type = HWConfigType.from_str(HW_CONFIG_TYPE_TARGET_DEVICE_MAP[target_device])
-    global_compression_lr_multiplier = config.get('compression_lr_multiplier', None)
-    global_ignored_scopes = config.get("ignored_scopes")
-    global_target_scopes = config.get("target_scopes")
+def extract_algo_specific_config(config: NNCFConfig, algo_name_to_match: str) -> Dict:
+    compression_section = config["compression"]
+    if isinstance(compression_section, list):
+        algo_list = compression_section
+    else:
+        assert isinstance(compression_section, dict)
+        algo_list = [compression_section]
 
-    if isinstance(compression_config_json_section, dict):
-        compression_config_json_section = [compression_config_json_section]
+    matches = []
+    for compression_algo_dict in algo_list:
+        algo_name = compression_algo_dict["algorithm"]
+        if algo_name == algo_name_to_match:
+            matches.append(compression_algo_dict)
 
-    compression_algorithm_configs = []
-    for algo_config in compression_config_json_section:
-        algo_config = NNCFConfig(algo_config)
-        algo_config.register_extra_structs(config.get_all_extra_structs_for_copy())
-        if hw_config_type is not None:
-            algo_config["hw_config_type"] = hw_config_type
-        if global_ignored_scopes is not None:
-            if "ignored_scopes" in algo_config:
-                algo_config["ignored_scopes"].extend(global_ignored_scopes)
-            else:
-                algo_config["ignored_scopes"] = global_ignored_scopes
-        if global_target_scopes is not None:
-            if "target_scopes" in algo_config:
-                algo_config["target_scopes"].extend(global_target_scopes)
-            else:
-                algo_config["target_scopes"] = global_target_scopes
-        if 'compression_lr_multiplier' not in algo_config:
-            algo_config['compression_lr_multiplier'] = global_compression_lr_multiplier
-        algo_config['target_device'] = target_device
-        compression_algorithm_configs.append(algo_config)
-
-    return compression_algorithm_configs
+    if len(matches) > 1:
+        raise RuntimeError(f"Multiple algorithm configurations specified for the same "
+                           f"algo {algo_name_to_match} in the NNCF config!")
+    if not matches:
+        raise RuntimeError(f"Did not find an algorithm configuration for "
+                           f"algo {algo_name_to_match} in the NNCF config!")
+    return next(iter(matches))
```

Comment:
Just want to ask: is this way to do `my_list[0]` is preferable to use?

Reply:
I'm not making anyone use it, but it seems more general for me.